lib/components/steps/Home.tsx

```diff
@@ -0,0 +1,107 @@
+import {Box, Button, Stack, Typography} from '@mui/material';
+import TABLE from '../../components/assets/svg/table.svg';
+import X from '../../components/assets/svg/x.svg';
+import {ModalLayout} from "../layout/ModalLayout.tsx";
+import StyledIconButton from '../common/StyledIconButton.tsx';
+import {useCdeContext} from "../../CdeContext.tsx";
+import {STEPS} from "../../models.ts";
+import {processMappingFile, validateMappingFile} from "../../services/csvService.ts";
+
+
+function Home() {
+    const {
+        setStep,
+        labName,
+        handleClose,
+        cdeFileMapping,
+        setLoadingMessage,
+        setErrorMessage,
+        setMapping
+    } = useCdeContext();
+
+
+    const handleStartMapping = async () => {
+        if (!cdeFileMapping) {
+            setErrorMessage('Mapping file not found')
+            return
+        }
+
+        setLoadingMessage('Validating and processing file...');
+
+        try {
+            const isValid = await validateMappingFile(cdeFileMapping);
+            if (isValid) {
+                const mapping = await processMappingFile(cdeFileMapping);
+                setMapping(mapping);
```

Comment:
This replaces the whole mapping in context with a new mapping. It might make sense to create a method to change a specific key in the mapping given the application use cases